rotkehlchen/chain/ethereum/modules/shutter/decoder.py

```diff
@@ -0,0 +1,123 @@
+                tx_log.topics[0] == b'\xdc\x17\xa5b\x85\x03\x17a\xfbz\x86\xd9\n1"u1 ^\x1e\x9e\x8f\x85\xf4\xd0\xf0:T\xea_\xb9v' and  # noqa: E501
+                tx_log.topics[1] == redeem_id
+            ):
+                vesting_contract_address = tx_log.address
+            elif (
+                tx_log.address == self.shu.evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == SHUTTER_AIDROP_CONTRACT and
+                hex_or_bytes_to_address(tx_log.topics[2]) == vesting_contract_address
+            ):
+                amount = token_normalized_value(token_amount=hex_or_bytes_to_int(tx_log.data), token=self.shu)  # noqa: E501
+                break
+        else:
+            log.error(f'Could not find the SHU transfer in {context.transaction.tx_hash.hex()}')
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.shu,
+            balance=Balance(amount=FVal(amount)),
+            location_label=hex_or_bytes_to_address(context.tx_log.topics[2]),
+            notes=f'Claim {amount} SHU from shutter airdrop into the vesting contract: {vesting_contract_address}',  # noqa: E501
+            counterparty=CPT_SHUTTER,
+            address=context.transaction.to_address,
+        ))
+
+    def _decode_delegation_change(self, context: DecoderContext) -> DecodingOutput:
+        """This contract function (delegateTokens), can only be called by the owner,
+        that is `context.transaction.from_address`. So not verifying the caller here."""
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[1])
+        delegator_note = ''
+        if delegator != context.transaction.from_address:
+            delegator_note = f' for {delegator}'
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.GOVERNANCE,
+            asset=self.shu,
+            balance=Balance(),
+            location_label=context.transaction.from_address,
+            notes=f'Change SHU Delegate{delegator_note} from {hex_or_bytes_to_address(context.tx_log.topics[2])} to {hex_or_bytes_to_address(context.tx_log.topics[3])}',  # noqa: E501
+            counterparty=CPT_SHUTTER,
+        ))
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            SHUTTER_AIDROP_CONTRACT: (self._decode_shutter_claim,),
+        }
+
+    def decoding_by_input_data(self) -> dict[bytes, dict[bytes, Callable]]:
```

Comment:
Why are you using decoding by input data? Decoding by input_data should be avoided as it's much wider-catching than the other rules, and as such also makes the entire decoding process slower.

WE only use it when we have no idea the contract that will generate the logs. As we do with SAFE. Since anyone can deploy a SAFE contract.

Here youdon't need this. You just need to use the `addresses_to_decoders` with `0xe485E2f1bab389C08721B291f6b59780feC83Fd7`  (shutter token address), with the `DelegateChanged` log.